core/src/smartcontracts/isi/triggers/set.rs

```diff
@@ -50,6 +49,8 @@ pub struct Set {
     by_call_triggers: HashMap<TriggerId, LoadedAction<ExecuteTriggerEventFilter>>,
     /// Trigger ids with type of events they process
     ids: HashMap<TriggerId, EventType>,
+    /// Original [`WasmSmartContract`]s by [`TriggerId`] for querying purposes.
+    original_contracts: HashMap<TriggerId, WasmSmartContract>,
```

Comment:
while this may be fine atm, this will waste a lot of memory. A proper thing to do would be to store a reference to the block (block height actually) where this trigger is stored like this:

```suggestion
    original_contracts: HashMap<TriggerId, usize>,
```

Reply:
also this would best be stored in `wsv` just like `WorldStateView::transactions`